conformance/tests/specialtypes_promotions.py

```diff
@@ -2,15 +2,32 @@
 Tests "type promotions" for float and complex when they appear in annotations.
 """
 
+from typing import assert_type
+
 # Specification: https://typing.readthedocs.io/en/latest/spec/special-types.html#special-cases-for-float-and-complex
 
-v1: float = 1
-v2: complex = 1.2
-v2 = 1
+v1: int = 1
+v2: float = 1
+v3: float = v1
+v4: complex = 1.2
+v4 = 1
 
 
 def func1(f: float):
-    f.numerator  # E
+    f.numerator  # E: attribute exists on int but not float
+
+    if isinstance(f, float):
+        f.hex()  # OK (attribute exists on float but not int)
+    else:
+        assert_type(f, int)
```

Comment:
Mypy doesn't pass this test, but that's not reflected in the automated test results (or the manual scoring, currently). The problem is that mypy treats this branch as unreachable, and it doesn't check `assert_type` statements in code that it considers unreachable. That means you can replace this line with `assert_type(f, <any other type>)`, and mypy will not emit an error here. Any ideas for how to write the test in a way that catches this issue in an automated way?

Reply:
Good catch. We could introduce a type error here and assert that type checkers show the error.